rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,424 @@
+import logging
+from http import HTTPStatus
+from typing import TYPE_CHECKING, Any, Literal, overload
+
+import requests
+
+from rotkehlchen.assets.asset import Asset, AssetWithOracles
+from rotkehlchen.constants import DAY_IN_SECONDS
+from rotkehlchen.constants.assets import A_USD
+from rotkehlchen.constants.prices import ZERO_PRICE
+from rotkehlchen.db.settings import CachedSettings
+from rotkehlchen.errors.asset import UnknownAsset, UnsupportedAsset
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.price import NoPriceForGivenTimestamp, PriceQueryUnsupportedAsset
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.externalapis.interface import ExternalServiceWithApiKeyOptionalDB
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.history.deserialization import deserialize_price
+from rotkehlchen.history.price import PriceHistorian
+from rotkehlchen.history.types import HistoricalPrice, HistoricalPriceOracle
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.interfaces import HistoricalPriceOracleInterface
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChainID, ExternalService, Price, Timestamp
+from rotkehlchen.utils.misc import ts_now
+from rotkehlchen.utils.mixins.penalizable_oracle import PenalizablePriceOracleMixin
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class Alchemy(
+    ExternalServiceWithApiKeyOptionalDB,
+    HistoricalPriceOracleInterface,
+    PenalizablePriceOracleMixin,
+):
```

Comment:
Do you need any change related to the api schema to accept this one? Have you checked deleting prices from the cache?

Reply:
not at all, the schemas use `HistoricalPriceOracleField` so that handles serialisation and deserialisation.